src/lints/manual/manual_assert.rs

```diff
@@ -0,0 +1,110 @@
+use cairo_lang_defs::{ids::ModuleItemId, plugin::PluginDiagnostic};
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::{db::SemanticGroup, Arenas, Expr, ExprIf, Statement};
+use cairo_lang_syntax::node::TypedStablePtr;
+use if_chain::if_chain;
+
+use crate::{
+    context::{CairoLintKind, Lint},
+    helper::is_panic_expr,
+    queries::{get_all_function_bodies, get_all_if_expressions},
+};
+
+pub struct ManualAssert;
+
+/// ## What it does
+///
+/// Checks for manual implementations of `assert` macro in an if expressions.
+///
+/// ## Example
+///
+/// ```cairo
+/// fn main() {
+///     let a = 5;
+///     if a == 5 {
+///         panic!("a shouldn't be equal to 5");
+///     }
+/// }
+/// ```
+///
+/// Can be rewritten as:
+///
+/// ```cairo
+/// fn main() {
+///     let a = 5;
+///     assert!(a != 5, "a shouldn't be equal to 5");
+/// }
+/// ```
+impl Lint for ManualAssert {
+    fn allowed_name(&self) -> &'static str {
+        "manual_assert"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "Manual assert detected. Consider using assert!() macro instead."
+    }
+
+    fn kind(&self) -> crate::context::CairoLintKind {
+        CairoLintKind::ManualAssert
+    }
+}
+
+pub fn check_manual_assert(
+    db: &dyn SemanticGroup,
+    item: &ModuleItemId,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let function_bodies = get_all_function_bodies(db, item);
+    for function_body in function_bodies.iter() {
+        let if_exprs = get_all_if_expressions(function_body);
+        let arenas = &function_body.arenas;
+        for if_expr in if_exprs.iter() {
+            check_single_manual_assert(db, if_expr, arenas, diagnostics);
+        }
+    }
+}
+
+fn check_single_manual_assert(
+    db: &dyn SemanticGroup,
+    if_expr: &ExprIf,
+    arenas: &Arenas,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let Expr::Block(ref if_block) = arenas.exprs[if_expr.if_block] else {
+        return;
+    };
+
+    // If there's an else block we ignore it.
+    if if_expr.else_block.is_some() {
+        return;
+    };
+
+    // Without tail.
+    if_chain! {
+        if if_block.statements.len() == 1;
```

Comment:
If the block has more than one statement it's still rewritable to code with `assert!`